src/pypowsybl_jupyter/networkmapwidget.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) 2024, RTE (http://www.rte-france.com)
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# SPDX-License-Identifier: MPL-2.0
+#
+
+import pathlib
+
+import anywidget
+import traitlets
+
+import json
+
+from ipywidgets import (
+    CallbackDispatcher
+)
+
+from pypowsybl.network import Network
+
+class NetworkMapWidget(anywidget.AnyWidget):
+    """
+    Creates a Network map widget, displaying substations and lines for a network. The widget allows zooming and pannign the map, filtering based on nominal voltages
+
+    Args:
+        network: the input network.
+        subId: if not None, centers the network on the substation with the given substation id. Default is None.
+        use_name: When True (default) the widget displays network's elements names (if available, otherwise their ids); When False, the widget displays network's elements ids.
+        display_lines: When True (default) the network lines are displayed on the map. When false, the widget displays only the substations.
+        use_line_extensions: When False (default) the widget does not use the network's line extensions; Each line is drawn as a straight line connecting two substations.
```

Comment:
`use_line_geodata` maybe to be clearer?
```suggestion
        use_line_geodata: When False (default) the widget does not use the network's line geodata extensions; Each line is drawn as a straight line connecting two substations.
```